crates/pep440-rs/src/version.rs

```diff
@@ -2320,11 +2412,32 @@ pub(crate) fn compare_release(this: &[u64], other: &[u64]) -> Ordering {
 ///
 /// [pep440-suffix-ordering]: https://peps.python.org/pep-0440/#summary-of-permitted-suffixes-and-relative-ordering
 fn sortable_tuple(version: &Version) -> (u64, u64, Option<u64>, u64, &[LocalSegment]) {
-    match (version.pre(), version.post(), version.dev(), version.min()) {
+    // If the version is a "max" version, use a post version larger than any possible post version.
+    let post = if version.is_max() {
+        // But don't convert a `dev` version to a `post` version. `1.0.dev0` should _not_ be greater
+        // than `1.0.post0.dev0`, since the latter means "a dev release of a post release of 1.0",
+        // not "a post release of a dev release of 1.0".
+        if version.is_dev() && !version.is_post() {
+            None
+        } else {
+            Some(u64::MAX)
+        }
+    } else {
+        version.post()
+    };
+
+    // If the version is a "max" version, use a local version larger than any possible local version.
+    let local = if version.is_max() {
+        &[LocalSegment::Number(u64::MAX)]
+    } else {
+        version.local()
+    };
```

Comment:
We could also consider incorporating `max` into the `match` below... This felt easier to get right.